core/src/main/java/com/microsoft/applicationinsights/TelemetryConfiguration.java

```diff
@@ -77,6 +80,33 @@ public static TelemetryConfiguration getActive() {
     }
 
     /**
+     * Set {@link TelemetryConfiguration} instance to be 'Active'.
+     * Telemetry configuration must not be set twice explicitly or through calling {@link TelemetryConfiguration#getActive()}.
+     *
+     * @param telemetryConfiguration preconfigured instance of Telemetry Configuration
+     */
+    public static void setActive(TelemetryConfiguration telemetryConfiguration) {
+        synchronized (s_lock) {
+            if (initialized) {
+                throw new IllegalStateException("Telemetry configuration is already initialized. " +
+                        "Please make sure that TelemetryConfiguration#setActive(TelemetryConfiguration) is not called twice " +
```

Comment:
Can we some how prevent this behavior programaticallly? If some one tries to call setActive() twice the second call is ignored?

Reply:
Answered in previous thread, this is probably an option, but I wouldn't implement it like that, it is too dangerous and hard to debug if something goes wrong, it should not be initialized more than once.